src/core/tests/Test_StatusHandlerTruncation.py

```diff
@@ -0,0 +1,784 @@
+        self.runtime.status_handler.log_truncated_patches()
+
+        # Assert complete status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.complete_status_file_path, 'r') as file_handle:
+            complete_substatus_file_data = json.load(file_handle)[0]
+
+        self.__assert_patch_summary_from_status(complete_substatus_file_data, Constants.ASSESSMENT, Constants.PATCH_ASSESSMENT_SUMMARY, Constants.STATUS_SUCCESS, patch_count, is_under_internal_size_limit=True, is_truncated=False)
+
+        # Assert no truncated status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.status_file_path, 'r') as file_handle:
+            substatus_file_data = json.load(file_handle)[0]
+
+        self.assertEqual(len(json.dumps(substatus_file_data).encode('utf-8')), len(json.dumps(complete_substatus_file_data).encode('utf-8')))  # Assert both files have same bytes
+        self.__assert_patch_summary_from_status(substatus_file_data, Constants.ASSESSMENT, Constants.PATCH_ASSESSMENT_SUMMARY, Constants.STATUS_SUCCESS, patch_count, is_under_internal_size_limit=True, is_truncated=False)
+
+        # Assert 'Count of patches removed from: [Assessment=xxx] [Installation=0] log message is called
+        self.__assert_truncated_patches_removed(complete_substatus_file_data, patch_count_assessment=patch_count)
+
+    def test_only_assessment_patches_over_size_limit_truncated(self):
+        """ Perform truncation on very large assessment patches and checks for time performance concern.
+        Before truncation: 100000 assessment patches in status
+        complete status file byte size: 19,022kb,
+        Expected (After truncation): 672 assessment patches in status
+        operation: Assessment,
+        assessment substatus name: PatchAssessmentSummary,
+        assessment substatus status: warning,
+        assessment errors code: 0 (success),
+        assessment errors details count: 0,
+        count of assessment patches removed: 99328,
+        truncated status file byte size: 126kb. """
+
+        self.runtime.execution_config.operation = Constants.ASSESSMENT
+        self.runtime.status_handler.set_current_operation(Constants.ASSESSMENT)
+
+        patch_count = 100000
+        self.__expected_truncated_patch_count = 672
+        test_packages, test_package_versions = self.__set_up_packages_func(patch_count)
+        self.runtime.status_handler.set_package_assessment_status(test_packages, test_package_versions, "Critical")
+        self.runtime.status_handler.set_assessment_substatus_json(status=Constants.STATUS_SUCCESS)
+        self.runtime.status_handler.log_truncated_patches()
+
+        # Assert complete status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.complete_status_file_path, 'r') as file_handle:
+            complete_substatus_file_data = json.load(file_handle)[0]
```

Comment:
Another non blocker: fetching json from a file can be extracted to a function and reused in all places where <>.complete.status and <>.status files are read.

Reply:
Add an ADO item for this one too

Reply:
single function is created

done.